### PR TITLE
Launchpad: Fix completion logic for "Customize welcome message" task

### PR DESCRIPTION
**Pull request by sixhours**
Related to https://github.com/Automattic/wp-calypso/issues/80503

## Proposed changes:

* Hook into the subscription option's add action in addition to the update action to ensure it completes the task whether adding or updating, not just on update.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply this patch to your sandbox and sandbox public-api
* Create a new site from `/setup/newsletter` and publish your first post on it
* On the Home screen, click on the "Customize welcome message" task
* Update the welcome message and save changes
* Go back to `/home` and the Customize welcome message task should be marked complete!

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php [hunk 1]
```diff
@@ -1390,19 +1390,35 @@ function wpcom_is_edit_page_task_visible() {
...
  * @return void
  */
-function wpcom_mark_customize_welcome_message_complete( $old_value, $value ) {
+function wpcom_mark_customize_welcome_message_complete_on_update( $old_value, $value ) {
 	if ( $value['invitation'] !== $old_value['invitation'] ) {
 		wpcom_mark_launchpad_task_complete( 'customize_welcome_message' );
 	}
 }
-add_action( 'update_option_subscription_options', 'wpcom_mark_customize_welcome_message_complete', 10, 3 );
+add_action( 'update_option_subscription_options', 'wpcom_mark_customize_welcome_message_complete_on_update', 10, 3 );
```
Comment (daledupreez) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php]: Nit: We're only looking at the first two arguments.
```suggestion
add_action( 'update_option_subscription_options', 'wpcom_mark_customize_welcome_message_complete_on_update', 10, 2 );
```

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php [hunk 2]
```diff
@@ -1390,19 +1390,35 @@ function wpcom_is_edit_page_task_visible() {
...
+ * @param string $value The value of the welcome message.
+ *
+ * @return void
+ */
+function wpcom_mark_customize_welcome_message_complete_on_add( $value ) {
+	if ( $value['invitation'] ) {
+		wpcom_mark_launchpad_task_complete( 'customize_welcome_message' );
+	}
+}
+add_action( 'add_option_subscription_options', 'wpcom_mark_customize_welcome_message_complete_on_add', 10, 3 );
```
Comment (daledupreez) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php]: Nit: We're only looking at the first argument.
```suggestion
add_action( 'add_option_subscription_options', 'wpcom_mark_customize_welcome_message_complete_on_add', 10, 1 );
```

**Review by daledupreez — DISMISSED** [review]
This is working as described, but I think we can clean up the hooks.

I did also need to manually add the task back into my newsletter task list, as we removed this task from the initial list yesterday.

Screenshot 2023-08-15 at 13 41 49

**Review by valterlorran — DISMISSED** [review]
LGTM! I think you only need to rebase it.

**Review by valterlorran — APPROVED** [review]
Still LGTM!

**Pull request merged**